src/libcore/slice/mod.rs

```diff
@@ -635,6 +639,18 @@ impl<T> SliceExt for [T] {
         self.binary_search_by(|p| p.borrow().cmp(x))
     }
 
+    fn rotate(&mut self, mid: usize) -> usize {
+        assert!(mid <= self.len());
+        let k = self.len() - mid;
+
+        unsafe {
+            let p = self.as_mut_ptr();
+            rotate::ptr_rotate(mid, p.offset(mid as isize), k);
+        }
+
+        k
```

Comment:
`self[..mid]` moves to `self[k..]` and `self[mid..]` moves to `self[..k]`, so when, say, rotating around a partition point, it gives the new partition point.

I return it because Stepanov returns the equivalent iterator, but I've definitely found the index less useful that the iterator.  Obviously it's easy to calculate yourself, but the more annoying problem is that I'm usually rotating a subslice, so the returned index needs to be fixed-up since it's not slice-independent.  (With iterators, in contrast, the new split point is non-trivial to calculate for non-random-access iterators, and iterators are independently-meaningful, so the returned split point can directly be used to construct ranges in the original container.)

So I don't have particularly strong feelings; let me know what you'd prefer.